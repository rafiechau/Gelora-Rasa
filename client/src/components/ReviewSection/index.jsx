import PropTypes from 'prop-types';
import StarIcon from '@mui/icons-material/Star';
import { FormattedMessage } from 'react-intl';
import classes from './style.module.scss';

const ReviewSection = ({ reviews }) => (
  <section className={classes.reviewSection} data-testid="review-section">
    {reviews.map((review, index) => (
      <div key={index} className={classes.reviewCard}>
        <div className={classes.cardContent}>
          <div className={classes.userInfo}>
            <img src={review.image} alt="User" className={classes.peopleReview} />
            <div className={classes.userDetails}>
              <h3 className={classes.userName}>{review.name}</h3>
              <p className={classes.userActivity}>{review.activity}</p>
              <div className={classes.userRating}>
                <StarIcon sx={{ color: 'yellow' }} />
                <span>
                  {review.rating}/5 <FormattedMessage id="app_stars" />
                </span>
              </div>
            </div>
          </div>
          <div className={classes.reviewText}>
            <h4 className={classes.reviewTitle}>{review.title}</h4>
            <p className={classes.reviewDescription}>{review.description}</p>
          </div>
        </div>
      </div>
    ))}
  </section>
);

ReviewSection.propTypes = {
  reviews: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      activity: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      rating: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default ReviewSection;
