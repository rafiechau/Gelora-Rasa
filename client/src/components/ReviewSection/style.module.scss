.reviewSection{
    background-color: lighten($navy, 5%);
    padding: 20px;
    margin-top: 2rem;
    .reviewCard{
        background-color: $navy;
        border-radius: 10px;
        box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
        display: flex;
        flex-direction: column; 
        align-items: center; 
        padding: 20px;
        margin: auto; 
        max-width: 800px; 
        @media (max-width: 768px) {
            max-width: 600px;
        }
    
        @media (max-width: 480px) {
            max-width: 100%;
            margin: 10px;
        }
        .cardContent{
            width: 100%;
            .userInfo {
                display: flex;
                align-items: center;
                gap: 15px; 
                @media (max-width: 768px) {
                    flex-direction: column;
                    align-items: flex-start;
                }
                .peopleReview {
                    width: 80px; 
                    height: 80px; 
                    border-radius: 50%;
                    object-fit: cover;
                    @media (max-width: 480px) {
                        width: 60px;
                        height: 60px;
                    }
                }
                .userName, .userActivity {
                    margin: 5px 0;
                }
                .userRating{
                    display: flex;
                    flex-direction: row;
                    align-items: center;
                }
            }
            .reviewText{
                .reviewTitle {
                    font-size: 1.2rem; 
                    @media (max-width: 768px) {
                        font-size: 1rem;
                    }
    
                    @media (max-width: 480px) {
                        font-size: 0.9rem;
                    }
                }
                  
                .reviewDescription {
                    color: $color-white-fade;
                    font-size: 0.95rem; 
                    margin-top: 10px;
                    @media (max-width: 768px) {
                        font-size: 1rem;
                    }
    
                    @media (max-width: 480px) {
                        font-size: 0.9rem;
                    }
                }
            }
            
        }
    }
}