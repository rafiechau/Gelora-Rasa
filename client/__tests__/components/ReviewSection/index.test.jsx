import React from 'react';
import { render, screen } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import ReviewSection from '@components/ReviewSection';

const mockReviews = [
  {
    name: 'John Doe',
    activity: 'Reviewed 2 days ago',
    image: 'path_to_image.jpg',
    rating: 4,
    title: 'Great Experience',
    description: 'I had a fantastic time using this product!',
  },
];

describe('ReviewSection', () => {
  beforeEach(() => {
    render(
      <IntlProvider locale="id" messages={{ app_stars: 'stars' }}>
        <ReviewSection reviews={mockReviews} />
      </IntlProvider>
    );
  });

  test('renders the review section', () => {
    expect(screen.getByTestId('review-section')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const reviewSection = screen.getByTestId('review-section');
    expect(reviewSection).toMatchSnapshot();
  });
});
