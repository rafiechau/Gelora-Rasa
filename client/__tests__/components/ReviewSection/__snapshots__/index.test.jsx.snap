// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewSection matches snapshot 1`] = `
<section
  class="reviewSection"
  data-testid="review-section"
>
  <div
    class="reviewCard"
  >
    <div
      class="cardContent"
    >
      <div
        class="userInfo"
      >
        <img
          alt="User"
          class="peopleReview"
          src="path_to_image.jpg"
        />
        <div
          class="userDetails"
        >
          <h3
            class="userName"
          >
            John Doe
          </h3>
          <p
            class="userActivity"
          >
            Reviewed 2 days ago
          </p>
          <div
            class="userRating"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-he82x6-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <span>
              4
              /5 
              stars
            </span>
          </div>
        </div>
      </div>
      <div
        class="reviewText"
      >
        <h4
          class="reviewTitle"
        >
          Great Experience
        </h4>
        <p
          class="reviewDescription"
        >
          I had a fantastic time using this product!
        </p>
      </div>
    </div>
  </div>
</section>
`;
